### ISSUE TITLE
[Android] Native EditText with TYPE_CLASS_NUMBER has focus issues

### ISSUE THREAD

**Issue by jonasbark** [flutter/flutter#86480]
Hi there,

Unfortunately, I have to resurface this issue:
https://github.com/flutter/flutter/issues/81029

While the input itself now works again the users still experience issues when another Flutter native TextField is involved. Interestingly it's only an issue with the TYPE_CLASS_NUMBER android input.

https://user-images.githubusercontent.com/1151304/125805015-d3340c4c-248d-4a50-aaa4-13ca2da00167.mp4

```
Flutter 2.4.0-5.0.pre.56 • channel master • https://github.com/flutter/flutter.git
Framework • revision 0d7e6798c9 (vor 30 Minuten) • 2021-07-15 15:20:46 +0200
Engine • revision 093c4ed1a3
Tools • Dart 2.14.0 (build 2.14.0-312.0.dev)

OnePlus 6T, Android 10
```
I would expect that switching back to the native view will let me input characters again.

Here's another scenario where the user cannot enter anything at all:
https://user-images.githubusercontent.com/1151304/125806178-b16fb708-9334-4294-8b25-2236e5bdb4c5.mp4

This is reproducible by having a FocusNode on the Flutter TextField and calling `unfocus` on it.

Here's a sample repository to showcase the issue:
https://github.com/jonasbark/flutterfocus

Related issue:
https://github.com/flutter-stripe/flutter_stripe/issues/14

**Comment by TahaTesser** [comment]
Hi @jonasbark 
Thank you for filing the issue and code sample, I can confirm the issue exist on multiple channels 

|  `stable`  | `master` |
| :---: |:---: |
|  ❌  |   ❌  | 

Check `flutter doctor -v` outputs for each channel below

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.3, on Microsoft Windows [Version 10.0.22000.65], locale en-US)
    • Flutter version 2.2.3 at C:\Users\Taha\Code\flutter_stable
    • Framework revision f4abaa0735 (2 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\android-sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\android-sdk
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31424.327
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] VS Code (version 1.58.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.24.0

[✓] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows       • windows-x64    • Microsoft Windows [Version 10.0.22000.65]
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 91.0.4472.124
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 91.0.864.67

! Doctor found issues in 1 category.
```

```console
[✓] Flutter (Channel master, 2.4.0-5.0.pre.76, on Microsoft Windows [Version 10.0.22000.65], locale en-US)
    • Flutter version 2.4.0-5.0.pre.76 at C:\Users\Taha\Code\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8cf1de3e68 (7 hours ago), 2021-07-16 01:11:02 -0400
    • Engine revision c2ffafea35
    • Dart version 2.14.0 (build 2.14.0-321.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\android-sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\android-sdk
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31424.327
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.2.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] VS Code (version 1.58.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.24.0

[✓] Connected device (5 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86     • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows       • windows-x64     • Microsoft Windows [Version 10.0.22000.65]
    • Windows (UWP) (desktop) • winuwp        • windows-uwp-x64 •
    • Chrome (web)            • chrome        • web-javascript  • Google Chrome 91.0.4472.124
    • Edge (web)              • edge          • web-javascript  • Microsoft Edge 91.0.864.67

• No issues found!
```
		
</details>

✅ : No Issue ❌: Issue reproduced

**Comment by chinmaygarde** [comment]
cc @HansMuller @goderbauer This is an issue where the text editing client is not being updated correctly.

**Comment by HansMuller** [comment]
Will ask @LongCatIsLooong to take a look

**Comment by LongCatIsLooong** [comment]
@blasten this seems to be platform views related. Would you like to take a look?

**Comment by daveols** [comment]
@blasten @LongCatIsLooong any updates on this one folks?

This issue essentially renders the native Stripe SDK unusable for Flutter apps wishing to preserve the PCI compliance of the underlying SDK.

**Comment by Lyba97** [comment]
Hi, is there any update on this issue? How can I use in my application. I am facing these focus issues and my application's deadline is critical.

**Comment by blasten** [comment]
This seems to be a regression since #81029 was fixed. Unfortunately, we don't have e2e tests for this scenario. 

We can probably now use the Android emulators added by @GaryQian to test this scenario, and prevent similar regressions.

**Comment by blasten** [comment]
It turns out this is not a regression. You can fix this issue by adding the following:

```diff
PlatformViewLink(
  viewType: viewType,
  surfaceFactory: (context, controller) =>
      AndroidViewSurface(
    controller: controller
        // ignore: avoid_as
        as AndroidViewController,
    gestureRecognizers: const <
        Factory<OneSequenceGestureRecognizer>>{},
    hitTestBehavior: PlatformViewHitTestBehavior.opaque,
  ),
  onCreatePlatformView: (params) {
    return PlatformViewsService.initSurfaceAndroidView(
      id: params.id,
      viewType: viewType,
      layoutDirection: TextDirection.ltr,
      creationParamsCodec: const StandardMessageCodec(),
+    onFocus: () {
+       params.onFocusChanged(true); // <-- Fix
+    } ,
    )
      ..addOnPlatformViewCreatedListener(
          params.onPlatformViewCreated)
      ..create();
  },
),
```

I just realized this isn't mentioned on https://flutter.dev/docs/development/platform-integration/platform-views#hybrid-composition.

Longer term, we should just update the implementation of `AndroidView`, so users don't have to wire these ups.

**Comment by jonasbark** [comment]
Hi @blasten!
I applied the change in the flutter_stripe repository but I don't see any change in the behavior. 
If you'd like to also have a look I prepared a test branch at https://github.com/flutter-stripe/flutter_stripe/tree/focus_fix.

In the example app open Card Payments > Simple - Using webhooks to reproduce it. Meanwhile, I'm also trying to figure out why it indeed does work in my flutterfocus repo but not for flutter_stripe.
